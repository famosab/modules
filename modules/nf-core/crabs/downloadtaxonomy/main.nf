process CRABS_DOWNLOADTAXONOMY {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/crabs:1.0.7--pyhdfd78af_0':
        'biocontainers/crabs:1.0.7--pyhdfd78af_0' }"

    input:
    val(meta)

    output:
    tuple val(meta), path("*/nucl_gb.accession2taxid"), emit: accession2taxid
    tuple val(meta), path("*/names.dmp")              , emit: names
    tuple val(meta), path("*/nodes.dmp")              , emit: nodes
    tuple val(meta), path("*")                        , emit: downloadtaxonomy_dir
    path "versions.yml"                               , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir ${prefix}

    crabs --download-taxonomy \\
        --output ${prefix} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        crabs: \$(crabs --help | grep 'CRABS |' | sed 's/.*CRABS | \\(v[0-9.]*\\).*/\\1/')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir ${prefix}
    touch ${prefix}/nucl_gb.accession2taxid
    touch ${prefix}/names.dmp
    touch ${prefix}/nodes.dmp

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        crabs: \$(crabs --help | grep 'CRABS |' | sed 's/.*CRABS | \\(v[0-9.]*\\).*/\\1/')
    END_VERSIONS
    """
}
