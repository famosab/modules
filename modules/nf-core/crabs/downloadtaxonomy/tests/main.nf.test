nextflow_process {

    name "Test Process CRABS_DOWNLOADTAXONOMY"
    script "../main.nf"
    process "CRABS_DOWNLOADTAXONOMY"

    tag "modules"
    tag "modules_nfcore"
    tag "crabs"
    tag "crabs/downloadtaxonomy"

    test("download") {

        when {
            params {
                module_args = "--exclude 'acc2taxid,taxdump'"
            }
            process {
                """
                input[0] = [ id:'test' ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("download - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [ id:'test' ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                    ).match() }
            )
        }

    }

}
