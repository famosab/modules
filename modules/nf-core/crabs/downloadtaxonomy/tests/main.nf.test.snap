{
    "download": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "accession2taxid": [
                    
                ],
                "names": [
                    
                ],
                "nodes": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-03T13:56:49.20584"
    },
    "download - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "nucl_gb.accession2taxid:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "names.dmp:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "nodes.dmp:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "names.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "nodes.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "nucl_gb.accession2taxid:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "versions.yml:md5,b98b46449a638f8825d42a6abe7a742e"
                        ]
                    ]
                ],
                "4": [
                    "versions.yml:md5,b98b46449a638f8825d42a6abe7a742e"
                ],
                "accession2taxid": [
                    [
                        {
                            "id": "test"
                        },
                        "nucl_gb.accession2taxid:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "downloadtaxonomy_dir": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                "names.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "nodes.dmp:md5,d41d8cd98f00b204e9800998ecf8427e",
                                "nucl_gb.accession2taxid:md5,d41d8cd98f00b204e9800998ecf8427e"
                            ],
                            "versions.yml:md5,b98b46449a638f8825d42a6abe7a742e"
                        ]
                    ]
                ],
                "names": [
                    [
                        {
                            "id": "test"
                        },
                        "names.dmp:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "nodes": [
                    [
                        {
                            "id": "test"
                        },
                        "nodes.dmp:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b98b46449a638f8825d42a6abe7a742e"
                ]
            },
            {
                "CRABS_DOWNLOADTAXONOMY": {
                    "crabs": "v1.0.7"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-03T14:07:27.923472"
    }
}